lib/ccflay.rb

```diff
@@ -84,3 +84,25 @@ def pure_ruby_hash # :nodoc: see above
     hash
   end
 end
+
+class Sexp
+  def flatter
+    r = dup.clear
+
+    each do |s|
+      if s.is_a? Sexp
+        ss = s.flatter
+
+        if ss.size == 2 && ss[1].is_a?(Sexp)
+          r.concat ss
+        else
+          r << ss
+        end
+      else
+        r << s
+      end
+    end
+
+    r
```

Comment:
I disagree with this for many reasons... but I think it doesn't matter because I'm gonna have to cache the old mass with `r.mass = mass` and that just doesn't fit with `each_with_object`.

Also... 15-18% slower?!? Yielding an extra parameter shouldn't cost THAT much.

```
Comparison:
            flatter1:        6.2 i/s
            flatter2:        5.3 i/s - 1.16x  slower
```